odoo/addons/base/models/ir_actions_report.py

```diff
@@ -111,6 +113,29 @@ class IrActionsReport(models.Model):
     attachment = fields.Char(string='Save as Attachment Prefix',
                              help='This is the filename of the attachment used to store the printing result. Keep empty to not save the printed reports. You can use a python expression with the object and time variables.')
 
+    @api.depends('model')
+    def _compute_model_id(self):
+        for action in self:
+            action.model_id = self.env['ir.model']._get(action.model).id
+
+    def _search_model_id(self, operator, value):
+        ir_model_ids = None
+        if isinstance(value, str):
+            names = self.env['ir.model'].name_search(value, operator=operator)
+            ir_model_ids = [n[0] for n in names]
+
+        if type(value) in (int, list, tuple):
```

Comment:
Use `isinstance`.
Better check if it's an iterable (https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable)
```suggestion
        elif isinstance(value, int):
            ir_model_ids = [value]
        elif isinstance(value, collections.abc.Iterable):
```